docs/ios/open-source/manual-instrumentation/custom-attributes.md

```diff
@@ -0,0 +1,224 @@
+    "device_category": UIDevice.current.userInterfaceIdiom == .pad ? "tablet" : "phone",
+    "app_install_date": installDateFormatter.string(from: appInstallDate)
+])
+```
+
+Permanent attributes can be removed when they're no longer needed:
+
+```swift
+Embrace.client?.removePermanentProperty(key: "legacy_setting")
+```
+
+## Span Attributes
+
+You can add attributes to spans to provide context specific to that operation:
+
+```swift
+let span = Embrace.client?.startSpan(name: "load_user_content")
+
+// Add attributes to the span
+span?.setAttribute(key: "content_type", value: "article")
+span?.setAttribute(key: "content_id", value: articleId)
+span?.setAttribute(key: "cached", value: isLoadingFromCache ? "true" : "false")
+
+// Your code here
+
+span?.end()
+```
+
+Span attributes are only attached to that specific span and won't affect other spans or the session as a whole.
+
+## Event Attributes
+
+When logging custom events (breadcrumbs), you can include attributes as properties:
+
+```swift
+Embrace.client?.logBreadcrumb(
+    name: "user_action",
+    properties: [
+        "action_type": "button_tap",
+        "target_screen": "settings",
+        "feature": "dark_mode"
+    ]
+)
+```
+
+## Common Use Cases
+
+### User Segments
+
+```swift
+// In your user service
+func updateUserSegments() {
+    let userSegments = [
+        "is_premium": userService.isPremiumUser ? "true" : "false",
+        "account_age_days": String(userService.accountAgeDays),
+        "has_completed_onboarding": userService.hasCompletedOnboarding ? "true" : "false"
+    ]
+    
+    Embrace.client?.addSessionProperties(userSegments)
+}
+```
+
+### Feature Flags
+
+```swift
+// Track enabled features
+func updateFeatureFlags() {
+    let enabledFeatures = featureFlagService.enabledFeatures.reduce(into: [String: String]()) { result, feature in
+        result[feature.flagName] = feature.isEnabled ? "true" : "false"
+    }
+    
+    Embrace.client?.addSessionProperties(enabledFeatures)
+}
+```
+
+### A/B Test Variants
+
+```swift
+// Track which A/B test variants the user is in
+func trackExperimentVariants() {
+    let experimentVariants = experimentService.activeExperiments.reduce(into: [String: String]()) { result, experiment in
+        let variantName = experiment.assignedVariant?.name ?? "control"
+        result["exp_\(experiment.name)"] = variantName
+    }
+    
+    Embrace.client?.addSessionProperties(experimentVariants)
+}
+```
+
+### Environment Information
+
+```swift
+func setupEnvironmentInfo() {
+    Embrace.client?.addPermanentProperties([
+        "environment": AppConfig.environment.rawValue,
+        "app_version": Bundle.main.infoDictionary?["CFBundleShortVersionString"] as? String ?? "unknown",
+        "build_number": Bundle.main.infoDictionary?["CFBundleVersion"] as? String ?? "unknown"
+    ])
+}
+```
+
+## Best Practices
+
+### Naming Conventions
+
+Use clear, descriptive names for attribute keys:
+- Use snake_case for consistency
```

Comment:
is this our recommendation across the board? i.e., do android, other platform, and dashboard users agree?